### ISSUE TITLE
Root word?

### ISSUE THREAD

**Issue by hgezim** [TarteelAI/quranic-universal-library#94]
[Bayan app](https://web.archive.org/web/20180422012859/https://bayan.io/) used to have word by word translations and you could find the root word for each Arabic word translation. Can these libs be used to build something like that?

**Comment by piraka9011** [comment]
Not directly, but you can TAL at https://www.quranmorphology.com/about and the linked sources

**Comment by mahmoodhamdi** [comment]
Yes — QUL ships exactly this kind of data, and you can build a Bayan-style word-by-word reader on top of it without running the CMS yourself.

Three resource families on the website map directly to that flow:

- **Quranic Morphology** — root, lemma, stem, POS per word: [https://qul.tarteel.ai/resources/morphology](https://qul.tarteel.ai/resources/morphology)
- **Word-by-word Translation** — translated text per word slot: [https://qul.tarteel.ai/resources/word-translation](https://qul.tarteel.ai/resources/word-translation) (or the canonical [tutorial-translation-end-to-end.md](https://github.com/TarteelAI/quranic-universal-library/blob/main/docs/tutorial-translation-end-to-end.md))
- **Quran Script (WBW glyph variants)** for the Arabic side: [https://qul.tarteel.ai/resources/quran-script](https://qul.tarteel.ai/resources/quran-script)

Each resource is downloadable as JSON or SQLite and joins on `surah_id` / `ayah_number` / `word_position`, which is documented in [`docs/data-model.md`](https://github.com/TarteelAI/quranic-universal-library/blob/main/docs/data-model.md). The [Morphology end-to-end tutorial](https://github.com/TarteelAI/quranic-universal-library/blob/main/docs/tutorial-morphology-end-to-end.md) walks through pulling root+lemma per word and showing it next to the Arabic text — that's the same building block Bayan used.

So no licensing/library question on the QUL side: download the three resource files above and you have what you need to render "Arabic word | translated word | root" inline.
